pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -1305,6 +1308,41 @@ protected DispatchRate internalGetSubscriptionDispatchRate() {
         return policies.subscriptionDispatchRate.get(pulsar().getConfiguration().getClusterName());
     }
 
+    public List<BundleStats> internalGetAllBundleStats() {
```

Comment:
Will this return stats for the bundles owned by this broker or will it return all bundle stats?

Reply:
this will return the bundles only owned by this broker, so we could known the traffic load of each bundle and decide whether to unload any bundles.

Reply:
Since it is only returning the stats for the namespace's bundles on this broker, I wonder if we should change the name to align with the returned value. When I first read the name of this endpoint, I assumed that it'd get the bundle stats for all bundles in the namespace. Do you know if we have a precedent for this kind of endpoint in the API already?

Reply:
Yes, we already have API `BrokerStats.getLoadReport()`, you can try with `bin/pulsar-admin broker-stats load-report`. The result contains all bundles' stats of this broker, but without topic list for each bundle. 
We could move the API of get all bundles' stats for this broker to `BrokerStats` to avoid misunderstanding. Keep the API of get single bundle stats in `Namespaces` and add lookup logic to ensure get the result even the single bundle is not owned by this broker. I will make this change if it's accecpt.